h/services/delete_user.py

```diff
@@ -23,17 +23,34 @@ def delete(self, user):
         message.
         """
 
-        if Group.created_by(self.request.db, user).count() > 0:
-            raise UserDeleteError('Cannot delete user who is a group creator.')
-
-        user.groups = []
+        # Check whether this user has created any groups that others have
+        # annotated in.
+        #
+        # We check for non-empty `group_ids` before querying the DB to avoid an
+        # expensive SQL query if `in_` is given an empty list (see
+        # https://stackoverflow.com/questions/23523147/)
+        group_ids = [g.pubid for g in user.groups]
```

Comment:
Umm… isn’t this going to check for any groups the user is a *member* of, rather than any groups they’ve *created*?

Reply:
🤦‍♂️ yes, it is.